consensus/dbft/dbft.go

```diff
@@ -895,8 +897,8 @@ func (c *DBFT) verifyPreBlockCb(b dbft.PreBlock[common.Hash]) bool {
 	}
 	ethBlock := dbftBlock.ToEthBlock()
 
-	staticPool := c.newStaticPool(parent)
-	errs := staticPool.Add(dbftBlock.transactions, false, false)
+	c.staticPool = newStaticPool(parent, c.chain)
```

Comment:
Move this initialization to `postBlock` callback prior to `handleDKG`, chain state at the current height should already be available by this moment. And here we need to reset only if CV happened:
```suggestion
	if c.dbft.Context.ViewNumber > 0 {
		c.staticPool.ResetStatic()
	}
```

The overall idea is to request state snapshot once per block.

Reply:
> Move this initialization to `postBlock` callback prior to `handleDKG`, chain state at the current height should already be available by this moment.

It looks a bit confusing to me, since we don't need the current state but the parent state to verify transactions.

If we are going to judge based on view number, then we can move the initialization into the `else` case here.

> The overall idea is to request state snapshot once per block.

OK, then let's only init it when `ViewNumber` is 0.